File: cpp/log-levels/log_levels.cpp
#include <string>

namespace LOG_LEVELS {
const std::string INFO = "[INFO]:";
const std::string WARNING = "[WARNING]:";
const std::string ERROR = "[ERROR]:";
} // namespace LOG_LEVELS

namespace log_line {
std::string message(std::string log) { return log.substr(log.find(":") + 2); }

std::string log_level(std::string log) {
  return log.substr(log.find('[') + 1, log.find(']') - 1);
}

std::string reformat(std::string log) {
  return message(log) + " (" + log_level(log) + ")";
}
} // namespace log_line

File: java/log-levels/src/main/java/LogLevels.java
public class LogLevels {
    public static String message(String logLine) {
        return logLine.substring(logLine.indexOf(":") + 2).strip();
    }

    public static String logLevel(String logLine) {
        return logLine.substring(logLine.indexOf("[") + 1, logLine.indexOf("]")).toLowerCase();
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine) + ")";
    }
}
